python/core/advanced-queues/heap-priority-queue-in-python.md

```diff
@@ -55,3 +55,48 @@ heapq.heapreplace(a, 999)
 print(a)
 # [7, 8, 999, 12, 9]
 ```
+
+--- 
+## Practice
+
+Replace the smallest number of the heap h with `50`:
+
+```python
+import heapq
+
+h = [1,2,3,4,5]
+heapq.???(h)
+
+???.???(h, 50)
+# 1
+```
+
+* heapify
+* heapq
+* heapreplace
+* h
+* replace
+* makeheap
+* create
+
+---
+## Revision
+
+What number will the following snippet remove?
+
+```python
+
+import heapq
+
+h = [1030, 239, 304, 320]
+heapq.heapify(h)
+
+heapq.replace(h, 100)
+???
+
```

Comment:
Can you create a code snippet that demonstrates this? I am not sure if this (and the practice) work this way.

Reply:
use pythontutor
http://pythontutor.com/

Reply:
then add the example to the links of this insight

Reply:
Hey, I managed to test both snippets.
There was indeed a mistake here, the method is not `replace`, but `heapreplace`. I added two links in the insight for users to get more familiarised with this concept.